pkg/util/object.go

```diff
@@ -0,0 +1,48 @@
+package util
+
+import "reflect"
+
+// VisitObjectStrings visits recursively all string fields in the object and call the
+// visitor function on them. The visitor function can be used to modify the
+// value of the string fields.
+func VisitObjectStrings(obj interface{}, visitor func(string) string) {
+	v := reflect.ValueOf(obj).Elem()
+
+	visit := func(val reflect.Value) {
+		switch val.Kind() {
+		case reflect.String:
+			if val.CanSet() {
+				val.SetString(visitor(val.String()))
+			}
+		case reflect.Ptr, reflect.Interface:
+			switch val.Elem().Kind() {
+			case reflect.Struct, reflect.Map:
+				VisitObjectStrings(val.Interface(), visitor)
+			}
+		case reflect.Slice, reflect.Array:
+			for c := 0; c < val.Len(); c++ {
+				VisitObjectStrings(val.Index(c).Addr().Interface(), visitor)
+			}
+		case reflect.Struct, reflect.Map:
+			if val.CanInterface() {
+				VisitObjectStrings(val.Addr().Interface(), visitor)
+			}
+		}
+	}
+
+	switch v.Kind() {
```

Comment:
Shouldn't you add a case for Arrays and Slices here as well and do the indexing (as you already do with Maps) instead of doing it in the visitor function? IMO indexing of any struct, built-in or not, should happen in the same place, either the visitor function or here. Actually it should probably happen here...